src/JsonOption/ExpressionConverter.cs

```
@@ -9,7 +9,15 @@ internal class ExpressionConverter : JsonConverter<Expression>
     {
         public override Expression Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            return new(reader.GetString(), "");
+            try
+            {
+                // parse to expression
+                return Expression.ToExpression(reader.GetString());
+            }
+            catch (Exception e)
+            {
+                return new(reader.GetString(), "");
```

In what case would that throw an exception? I feel like catching everything and discarding the exception could be dangerous...

---

Its a simple fallback if something goes wrong in the paring of the string. Even if its not my fault, e.g. something changed on the api. any ideas to handle it better?

---

Well in the case of something changing in the API I think it would be better to just rethrow so we could be aware of it and fix it

---

I am afraid of breaking some existing software, because it is "just a feature" no one is using for now and we could not test every case by definition (academic licence)! If we throw an exception the users are damned to use this feature correctly.
And if something goes wrong it breaks the JsonConverter which is in our hands and the user had no access to.

I can think of handling it within the ToExpression method express something like "HasErrors" or "HasParsingErrors" or/and rename it to TryParse ... so its less "demanding" 😬

---

Yeah maybe doing a TryParse and keeping unit tests could be a solution?

---

I like it, cause nothing can break and the error is somehow expressed to the user.

---

had another idea. we do not parse the expression by default! (keep everything like before) So the user can decide to do so and is in charge to handle exceptions. nothing breaks, the feature stands alone, no potential performance impact on exceptions.

P.S.: i tried the try parse pattern and it doesnt really fit for the JsonConvert and would be very clunky.

---

Ah yeah could be a good idea too